Sources/StreamChat/StateLayer/MemberList.swift

```diff
@@ -0,0 +1,86 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// An object which represents a list of `ChatChannelMember` for the specified channel.
+@available(iOS 13.0, *)
+public final class MemberList {
+    private let query: ChannelMemberListQuery
+    private let memberListUpdater: ChannelMemberListUpdater
+    private let stateBuilder: StateBuilder<MemberListState>
+    
+    init(query: ChannelMemberListQuery, client: ChatClient, environment: Environment = .init()) {
+        self.query = query
+        memberListUpdater = environment.memberListUpdaterBuilder(
+            client.databaseContainer,
+            client.apiClient
+        )
+        stateBuilder = StateBuilder {
+            environment.stateBuilder(
+                query,
+                client.databaseContainer
+            )
+        }
+    }
+    
+    /// An observable object representing the current state of the member list.
+    @MainActor public lazy var state: MemberListState = stateBuilder.build()
+    
+    /// Fetches the most recent state from the server and updates the local store.
+    ///
+    /// - Throws: An error while communicating with the Stream API.
+    public func get() async throws {
+        let pagination = Pagination(pageSize: .channelMembersPageSize)
+        try await loadMembers(with: pagination)
+    }
+    
+    /// Loads channel members for the specified pagination parameters and updates ``MemberListState/members``.
+    ///
+    /// - Parameter pagination: The pagination configuration which includes a limit and an offset or a cursor.
+    ///
+    /// - Throws: An error while communicating with the Stream API.
+    /// - Returns: An array of channel members for the pagination.
+    @discardableResult public func loadMembers(with pagination: Pagination) async throws -> [ChatChannelMember] {
+        try await memberListUpdater.load(query.withPagination(pagination))
+    }
+    
+    /// Loads more channel members and updates ``MemberListState/members``.
+    ///
+    /// - Parameters
+    ///   - limit: The limit for the page size. The default limit is 30.
+    ///
+    /// - Throws: An error while communicating with the Stream API.
+    /// - Returns: An array of channel members.
+    @discardableResult public func loadMoreMembers(limit: Int? = nil) async throws -> [ChatChannelMember] {
+        let pageSize = limit ?? Int.channelMembersPageSize
```

Comment:
We should also use the query pagination size here too if the limit is nil

Reply:
I'll go over all the cases with one commit. Will review them all.